app/lib/dns.server.ts

```diff
@@ -169,14 +152,13 @@ export const getChangeStatus = async (changeId: string) => {
     }
     return response.ChangeInfo.Status;
   } catch (error) {
-    // TODO: fix with better logging...
-    console.warn(error);
+    logger.warn(error);
     throw new Error(`Error occurred while getting change status`);
   }
 };
 
 const isNameValid = (name: string) => {
```

Comment:
Add a comment above this function that indicates what our criteria is for a valid name in plain language (i.e., so people don't have to parse your regex to understand it.

Also, according to [Route53's docs](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DomainNameFormat.html), we should be doing this:

> For domain name registration, a domain name can contain only the characters a-z, 0-9, and - (hyphen). You can't specify a hyphen at the beginning or end of a label....Amazon Route 53 stores alphabetic characters as lowercase letters (a-z), regardless of how you specify them: as uppercase letters, lowercase letters, or the corresponding letters in escape codes.

So from this I think we want:

- can't start or end with `-`
- can use `A-Za-z0-9-` (upper case is fine, but we'll convert to lower)
- can have multiple sections like this separated by `.`

What about this (untested):

```ts
/^[a-z0-9]([a-z0-9-]+\.?)+[a-z0-9]$/i
```

Reply:
I would probably use an external library for domain name validation, if we want to validate it ourselves, I.e.,
https://www.npmjs.com/package/validator ==> isFQDN

Reason: it is extremely hard to reliably validate a domain name, specially if I take stuff like internaltional characters in the domain name (i.e. punycode coded domain segment) and the regexes that does so are usually a mile long

Reply:
Agree, it's hard to validate all possible valid domains.  I'd argue we should further limit this (e.g., I don't think we should allow punycode etc. for our use case, since it has the potential to open us to abuse).

Reply:
there are so many things that is hard to validate i.e.two dashes are invalid, two dots next to each other is invalid, Do we want to offer multi level subdomains, i.e. `a.b.studentid.senecastudent.ca`, because if we do, that's not covered with the wildcard SSL cert IIRC, underscoreas also dowss have limitations IIRC

I'd probably change the signature of this:

const fn = {{subdomain ...}} => {
  // validate a single level of subdomain
  const fqdn = `${subdomain}.${process.env.PARENT_DOMAIN}`
  //  create subdomain
}

Reply:
Forcing them to fit in the pattern `{name}.{username}.senecastudent.ca` is probably good, agree.  And let's keep the list of what's allowed in `name` to something pretty small: `a-zA-Z0-9-`.

Reply:
If we allow `-` in the name, we need to also make sure to not start with it, and don't have multiple in a sequence like `--`

We will also need _ because of the let's encrypt challenge:
`_acme-challenge.{username}.senecastudent.ca`

Reply:
I opened a new issue. #185 I would like to close this this PR now. @dadolhay I will soon write another PR to improve regex part.

Reply:
Perfect.  Good idea to separate the regex bit so we don't block all this good work on one issue.